docs/Index.tsx

```diff
@@ -0,0 +1,371 @@
+}
+
+interface IMemberEditViewState extends IMemberEditViewStateProps {
+  modelCid?: string;
+}
+
+const actions = {
+  AddMemberEditView: 'ADD_MEMBER_EDIT_VIEW',
+  ToggleOpenedTetherElement: 'TOGGLE_OPENED_TETHER_ELEMENT',
+  OnChangeMemberEmail: 'ON_CHANGE_MEMBER_EMAIL',
+  OnChangeMemberSendEmail: 'ON_CHANGE_MEMBER_SEND_EMAIL',
+  CancelMemberEdition: 'ON_CANCEL_MEMBER_EDITION'
+};
+
+const memberReducers = (state: IMemberEditViewState = { isOpen: false }, action): IMemberEditViewState => {
+  let newState = _.extend({}, state);
+  switch (action.type) {
+    case actions.AddMemberEditView:
+      newState.modelCid = action.modelCid;
+      return newState;
+    case actions.ToggleOpenedTetherElement:
+      if (action.modelCid == state.modelCid) {
+        newState.isOpen = action.isOpen;
+        return newState;
+      }
+      return state;
+    case actions.OnChangeMemberEmail:
+      if (action.modelCid == state.modelCid) {
+        newState.email = action.email;
+        return newState;
+      }
+      return state;
+    case actions.OnChangeMemberSendEmail:
+      if (action.modelCid == state.modelCid) {
+        newState.sendEmail = action.sendEmail;
+        return newState;
+      }
+      return state;
+    case actions.CancelMemberEdition:
+      if (action.modelCid == state.modelCid) {
+        return {
+          isOpen: false,
+          modelCid: state.modelCid,
+          email: action.email,
+          sendEmail: action.sendEmail
+        };
+      }
+      return state;
+    default:
+      return state;
+  }
+};
+
+const mapStateToProps = (state: IMemberEditViewState[], ownProps: IMemberEditViewPropsConnected): IMemberEditViewStateProps => {
+  let item = _.find(state, (memberState: IMemberEditViewState) => {
+    return memberState.modelCid == ownProps.memberModel.cid;
+  });
+
+  return {
+    isOpen: item && !_.isUndefined(item.isOpen) ? item.isOpen : false,
+    email: item && !_.isUndefined(item.email) ? item.email : ownProps.memberModel.email,
+    sendEmail: item && !_.isUndefined(item.sendEmail) ? item.sendEmail : ownProps.memberModel.sendEmail
+  };
+};
+
+const addMember = (modelCid: string) => {
+  return {
+    type: actions.AddMemberEditView,
+    modelCid
+  };
+};
+
+const toggleOpenedTetherElement = (modelCid: string, isOpen: boolean) => {
+  return {
+    type: actions.ToggleOpenedTetherElement,
+    modelCid,
+    isOpen
+  };
+};
+
+const onChangeEmail = (modelCid: string, email: string) => {
+  return {
+    type: actions.OnChangeMemberEmail,
+    modelCid,
+    email
+  };
+};
+
+const onChangeSendEmail = (modelCid: string, sendEmail: boolean) => {
+  return {
+    type: actions.OnChangeMemberSendEmail,
+    modelCid,
+    sendEmail
+  };
+};
+
+const onClickCancel = (modelCid: string, email: string, sendEmail: boolean) => {
+  return {
+    type: actions.CancelMemberEdition,
+    modelCid,
+    email,
+    sendEmail
+  };
+};
+
+const mapDispatchToProps = (dispatch, ownProps) => {
```

Comment:
If you want to type this, ownProps has `IMemberEditViewPropsConnected` as its type and dispatch has `(action: Redux.Action) => void` as its type.